packages/app/src/setupTests.ts

```diff
@@ -19,3 +19,9 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom';
+
+// Do not remove, patching jsdom environment to support TextEncoder, refer to https://github.com/jsdom/jsdom/issues/2524
+// eslint-disable-next-line no-restricted-imports
+import { TextEncoder } from 'util';
+
+global.TextEncoder = TextEncoder;
```

Comment:
since this is very plugin specific, you could move it in your tests:

```ts
  beforeAll(() => {
    originalTextEncoder = global.TextEncoder;
    global.TextEncoder = TextEncoder;
  });

  afterAll(() => {
    global.TextEncoder = originalTextEncoder;
  });
  ```

Reply:
There's an example of this kind of fix in the `setupTests.ts` from the kubernetes plugin.

https://github.com/backstage/backstage/blob/master/plugins/kubernetes/src/setupTests.ts

Reply:
but here we are not in the kubernetes plugin, right?  So why should `setupTests.ts` of another package be changed? 🤔 Can we avoid that? Otherwise the same needs to be done in another plugin, in case they use code exposed by the k8s plugin

Reply:
This has been an issue with JSDom for a long time, we're just working around it, we should have this fixed for the full project, if only there was a central `setupTests.ts` for the entire repo to put this fix in...

Reply:
is it possible to fix it in another way? I guess the issue is caused by invoking `new TextEncoder();` globally, instead at runtime, in the functions where is needed

Reply:
TextEncoder is supposed to be a globally available API in both browser & NodeJS, it's just that JSDom doesn't provide it as it should, doesn't matter where you use it, this workaround is only for the test, the code doesn't need this.

The biggest issue here would be that the testing setup we have in Backstage is modularized, meaning each package/plugin has it's own testing setup, if I were to think of a way to apply this in the whole repo, it would be in the backstage CLI, would still touch a separate package, and is riskier since it could affect the whole project, if you prefer that let me know.

Reply:
@vinzscam @luchillo17  I think you two are maybe driving in different directions, where @vinzscam is hoping to minimize the number of affected packages but @luchillo17 is intentionally trying to affect all packages (so they benefit from a workaround for a silly bug) in a future-proof fashion.

I'm inclined to support @luchillo17's approach -- the bug in question is jsdom/jsdom#2524. It's been a while since I dove into that discussion, but my impression was that the best strategy is to just apply a blanket workaround like this and forget about it.

Reply:
Yes, that's exactly right, It has been years with this JSDom bug, and this is basically the official workaround.

If I could I would apply it to the whole repo, but I don't want to go to the CLI package to see if there's actually a `common` setup file for all packages & plugins.